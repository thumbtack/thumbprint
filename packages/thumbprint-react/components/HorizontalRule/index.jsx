import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './index.module.scss';

const propTypes = {
    /**
     * Renders the horizontal rule with the thumbtack styles.
     */
    lineStyle: PropTypes.oneOf(['solid', 'dotted', 'dashed']),
    color: PropTypes.oneOf(['gray', 'gray-300']),
};

const defaultProps = {
    lineStyle: 'solid',
    color: 'gray',
};

export default function HorizontalRule({ lineStyle, color }) {
    return (
        <hr
            className={classNames({
                [styles.root]: true,
                [styles.gray300]: color === 'gray-300',
                [styles.dashed]: lineStyle === 'dashed',
                [styles.dotted]: lineStyle === 'dotted',
            })}
        />
    );
}

HorizontalRule.propTypes = propTypes;
HorizontalRule.defaultProps = defaultProps;
