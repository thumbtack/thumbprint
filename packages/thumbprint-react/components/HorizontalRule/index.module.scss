@import '~@thumbtack/thumbprint-tokens/dist/scss/_index';

.root {
    border: 0;
    height: 0;
    border-top: 1px solid $tp-color__gray;
    margin-top: $tp-space__2 - 1px;
    margin-bottom: $tp-space__2;
}

.dotted {
    border-top-style: dotted;
}

.dashed {
    border-top-style: dashed;
}

.gray300 {
    border-top-color: $tp_color__gray-300;
}
