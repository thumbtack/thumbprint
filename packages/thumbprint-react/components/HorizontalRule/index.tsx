import React from 'react';
import classNames from 'classnames';
import styles from './index.module.scss';

interface PropTypes {
    /**
     * Controls the style of the line rendered for the horizontal rule.
     */
    lineStyle?: 'solid' | 'dotted' | 'dashed';
    /**
     * Controls the color of the line rendered for the horizontal rule.
     */
    color?: 'gray' | 'gray-300';
}

export default function HorizontalRule({
    lineStyle = 'solid',
    color = 'gray',
}: PropTypes): JSX.Element {
    return (
        <hr
            className={classNames({
                [styles.root]: true,
                [styles.gray300]: color === 'gray-300',
                [styles.dashed]: lineStyle === 'dashed',
                [styles.dotted]: lineStyle === 'dotted',
            })}
        />
    );
}
