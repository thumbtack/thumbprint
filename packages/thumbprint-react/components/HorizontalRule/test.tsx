import React from 'react';
import { mount } from 'enzyme';
import HorizontalRule from './index';

test('renders default horizontal rule', () => {
    const wrapper = mount(<HorizontalRule />);
    expect(wrapper).toMatchSnapshot();
});

test('renders dotted horizontal rule', () => {
    const wrapper = mount(<HorizontalRule lineStyle="dashed" />);
    expect(wrapper).toMatchSnapshot();
});

test('renders gray-300 horizontal rule', () => {
    const wrapper = mount(<HorizontalRule color="gray-300" />);
    expect(wrapper).toMatchSnapshot();
});
