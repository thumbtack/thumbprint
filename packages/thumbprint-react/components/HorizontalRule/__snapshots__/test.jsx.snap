// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default horizontal rule 1`] = `
<HorizontalRule
  color="gray"
  lineStyle="solid"
>
  <hr
    className="root"
  />
</HorizontalRule>
`;

exports[`renders dotted horizontal rule 1`] = `
<HorizontalRule
  color="gray"
  lineStyle="dashed"
>
  <hr
    className="root dashed"
  />
</HorizontalRule>
`;

exports[`renders gray-300 horizontal rule 1`] = `
<HorizontalRule
  color="gray-300"
  lineStyle="solid"
>
  <hr
    className="root gray300"
  />
</HorizontalRule>
`;
